app/src/main/java/org/mozilla/fenix/browser/UserAgentInterceptor.kt

```diff
@@ -0,0 +1,36 @@
+package org.mozilla.fenix.browser
+
+import mozilla.components.concept.engine.EngineSession
+import mozilla.components.support.ktx.kotlin.sha1
+
+/**
+ * An interceptor that updates the User-Agent of a request.
+ *
+ * Follow up: https://github.com/mozilla-mobile/fenix/issues/3341
+ */
+object UserAgentInterceptor {
+
+    /**
+     * Updates the User-Agent based on the [whitelistUaFilter] to set the [USER_AGENT_GECKO_69_0] value.
+     */
+    fun updateUserAgent(session: EngineSession, host: String) {
+        session.settings.userAgentString = if (whitelistUaFilter.contains(host.sha1())) {
+            USER_AGENT_GECKO_69_0
+        } else {
+            null
+        }
+    }
+
+    /**
+     * The white-listed domains.
+     */
+    private val whitelistUaFilter = setOf(
+        "1b12ecd917215146f79a0ac5e01b3059faadab47",
+        "a486f819018512f60a8a66324e51be0e1118a91d"
+    )
+
+    /**
+     * The User-Agent to use for the white-listed domains.
+     */
+    private const val USER_AGENT_GECKO_69_0 = "Mozilla/5.0 (Android 9; Mobile; rv:69.0) Gecko/69.0 Firefox/69.0"
```

Comment:
We should use "... Android ${android.os.Build.VERSION.RELEASE}" ... instead of hard-coding to `9`.

Reply:
Oops!